src/identity.jl

```diff
@@ -166,3 +168,5 @@ LinearAlgebra.I(n::Int, N::DataType=Float64) = IdentityMultiple(one(N)*I, n)
 # callable identity matrix given the scaling factor and the size
 IdentityMultiple(λ::Number, n::Int) = IdentityMultiple(λ*I, n)
 LinearAlgebra.I(λ::Number, n::Int) = IdentityMultiple(λ*I, n)
+
+LinearAlgebra.Hermitian(𝐼::IdentityMultiple) = Hermitian(convert(Matrix, 𝐼))
```

Comment:
It should be more efficient to create a `Diagonal` matrix instead of a normal `Matrix`.
```suggestion
function LinearAlgebra.Hermitian(𝐼::IdentityMultiple)
    return Hermitian(Diagonal(fill(𝐼.M.λ, 𝐼.n)))
end
```

Reply:
This definition is ok, but i'm not sure it is the most convenient one, since it is not exploiting the structure of the identity multiple. From a quick look at the source code of [diagonal.jl](https://github.com/JuliaLang/julia/blob/2d5741174ce3e6a394010d2e470e4269ca54607f/stdlib/LinearAlgebra/src/diagonal.jl#L114), some functions `ishermitian` are defined, and other specifics eg. [cholesky!](https://github.com/JuliaLang/julia/blob/2d5741174ce3e6a394010d2e470e4269ca54607f/stdlib/LinearAlgebra/src/diagonal.jl#L646), which should be specialized upon `IdentityMultiple` ... 

If you want, we can keep the version proposed by @schillic here and open a follow-up issue to add such methods.